File: offer/pdd/2.cpp
#include <vector>
#include <iostream>
#include <string>
#include <unordered_map>

using namespace std;

int main(){
    int N;
    cin>>N;
    vector<int> ddj(N),ddy(N);
    for(int i=N-1;i>=0;i--){
        int t;
        cin>>t;
        ddj[i] = t;
    }
    for(int i=N-1;i>=0;i--){
        int t;
        cin>>t;
        ddy[i] = t;
    }
    unordered_map<int,int> map;
    vector<int> table;
    int cddj = 0,cddy = 0;
    while (ddj.size()>0||ddy.size()>0)
    {
        if(ddj.size()>0){
            int t = ddj[ddj.size()-1];
            ddj.pop_back();
            if(map[t]){
                while (table.size()>0&&table[table.size()-1]!=t)
                {
                    table.pop_back();
                    cddj++;
                }
                cddj+=2;
                if(ddj.size()>0){
                    t = ddj[ddj.size()-1];
                    ddj.pop_back();
                    tab
                }
            }

        }
        if(map[table.back()]){
            while (table.back())
            {
                /* code */
            }
            
        }
    }
    
}

File: offer/pdd/2.py
N = int(input())
ddj = input().split(" ")
ddy = input().split(" ")

for i in range(N):
    ddj[i] = int(ddj[i])
    ddy[i] = int(ddy[i])
cddj = 0
cddy = 0
table = []
flag = False
while len(ddj)>0 or len(ddy)>0:
    if flag:
        flag = False
        if len(ddy)>0:
            t = ddy[0]
            ddy.pop(0)
            if t in table:
                while table[-1]!=t and len(table)>0:
                    cddy+=1
                    table.pop()
                table.pop()
                cddy+=2
                if len(ddy)>0:
                    continue
            else:
                table.append(t)
    else:
        if len(ddj)>0:
            t = ddj[0]
            ddj.pop(0)
            if t in table:
                while table[-1]!=t and len(table)>0:
                    cddj+=1
                    table.pop()
                table.pop()
                cddj+=2
                if len(ddj)>0:
                    continue
            else:
                table.append(t)
        if len(ddy)>0:
            t = ddy[0]
            ddy.pop(0)
            if t in table:
                while table[-1]!=t and len(table)>0:
                    cddy+=1
                    table.pop()
                table.pop()
                cddy+=2
                if len(ddy)>0:
                    table.append(ddy[0])
                    ddy.pop(0)
            else:
                table.append(t)
for i in table:
    if i%2==1:
        cddj+=1
    else:
        cddy+=1
print(cddj,cddy)



